Makefile

```diff
@@ -85,37 +97,28 @@ gen/.build:
 
 base.c: $(CONF)
 
-$(DEPS): $(SRCS)
```

Comment:
In Fact, during the development i getted some dependency errors due to relocation of openssl and pfvar.
And this parts doesn't work, it getting me :
```
Compile with OpenSSL by default. To compile with PolarSSL, run 'make USE_CRYPTO_POLARSSL=true' instead.
.depend:1: *** missing separator.  Stop.
```
clang (on mac, `clang` is the default `$(CC)`) doesn't find dependency and making error, so the command launch the `for` loop code.
so the `.depend` file looked like that :
```
$ cat .depend 
-n
-n parser.o: 
parser.c
-n main.o: 
main.c
-n redsocks.o: 
redsocks.c
-n log.o: 
log.c
-n direct.o: 
direct.c
-n ipcache.o: 
ipcache.c
-n autoproxy.o: 
autoproxy.c
-n encrypt.o: 
encrypt.c
-n shadowsocks.o: 
shadowsocks.c
-n http-connect.o: 
http-connect.c
-n socks4.o: 
socks4.c
-n socks5.o: 
socks5.c
-n http-relay.o: 
http-relay.c
-n base.o: 
base.c
-n base64.o: 
base64.c
-n md5.o: 
md5.c
-n http-auth.o: 
http-auth.c
-n utils.o: 
utils.c
-n redudp.o: 
redudp.c
-n socks5-udp.o: 
socks5-udp.c
-n shadowsocks-udp.o: 
shadowsocks-udp.c
-n tcpdns.o: 
tcpdns.c
-n gen/version.o: 
gen/version.c
```
So in first time, i removed this complicated loop because it's not working and i don't understand the point of this code.
and then i add `-I$(OSX_HEADERS_PATH)` to get it working.
`$(CC) -MM -I$(OSX_HEADERS_PATH) $(SRCS) 2>/dev/null >$(DEPS) || \`
```
$ cat .depend 
parser.o: parser.c utils.h parser.h log.h
main.o: main.c log.h main.h parser.h utils.h version.h
redsocks.o: redsocks.c list.h utils.h parser.h log.h main.h base.h \
  redsocks.h
log.o: log.c utils.h log.h
direct.o: direct.c parser.h log.h main.h redsocks.h list.h utils.h
ipcache.o: ipcache.c redsocks.h list.h utils.h log.h parser.h main.h \
  ipcache.h
autoproxy.o: autoproxy.c utils.h log.h redsocks.h list.h parser.h main.h \
  ipcache.h
encrypt.o: encrypt.c encrypt.h
shadowsocks.o: shadowsocks.c utils.h log.h redsocks.h list.h encrypt.h \
  shadowsocks.h
http-connect.o: http-connect.c log.h redsocks.h list.h utils.h \
  http-auth.h
socks4.o: socks4.c log.h redsocks.h list.h utils.h
socks5.o: socks5.c utils.h log.h redsocks.h list.h socks5.h
http-relay.o: http-relay.c log.h redsocks.h list.h utils.h http-auth.h
base.o: base.c config.h xnu/10.12/net/pfvar.h xnu/10.12/libkern/tree.h \
  xnu/10.12/net/radix.h log.h main.h parser.h redsocks.h list.h utils.h
base64.o: base64.c utils.h base64.h
md5.o: md5.c md5.h
http-auth.o: http-auth.c md5.h base64.h http-auth.h
utils.o: utils.c config.h main.h parser.h log.h base.h utils.h redsocks.h \
  list.h libc-compat.h
redudp.o: redudp.c list.h utils.h log.h socks5.h parser.h main.h \
  redsocks.h redudp.h libc-compat.h
socks5-udp.o: socks5-udp.c main.h parser.h utils.h log.h redudp.h list.h \
  redsocks.h socks5.h
shadowsocks-udp.o: shadowsocks-udp.c utils.h log.h redsocks.h list.h \
  main.h parser.h redudp.h encrypt.h shadowsocks.h
tcpdns.o: tcpdns.c list.h utils.h log.h parser.h main.h redsocks.h \
  tcpdns.h
version.o: gen/version.c gen/../version.h
```

But the -I is OS dependent, and this include is already done with $(CFLAGS)
So it's why i add $(CFLAGS) in this command : 
```
$(CC) -MM $(CFLAGS) $(SRCS) 2>/dev/null >$(DEPS)
```
But here i'm less confident on the good practices ^^

So to answer, I  test it on MacOS and Ubuntu virtual machine. it compile well.
I don't get the point of getting dependency to making a C project without cc. The blame on this loop give me this commit : https://github.com/semigodking/redsocks/commit/88f97036034e1c2ccf5d9d61949814ccf77fed70 with this reason `- Create file w/ dependencies even if there is no gcc. (portability)`
But i don't know well other OS, maybe it's for other compiler.
And i finally understand why the loop is broken. The default shell on MacOS is /bin/sh and the builtin echo of /bin/sh doesn't have arguments. so `echo -n` will display `-n`.